nx/exercises/exercises-1-20.livemd

```diff
@@ -0,0 +1,174 @@
+# Exercises: 1-20
+
+```elixir
+Mix.install([{:nx, "~> 0.6"}])
+```
+
+## Introduction
+
+Inspired by the Python notebook _100 Numpy Exercises_:
+
+https://www.kaggle.com/code/utsav15/100-numpy-exercises/notebook
+
+## 1-10
+
+**1. Install Nx in a Livebook (★☆☆)**
+
+```elixir
+Mix.install([{:nx, "~> 0.6"}])
+```
+
+**2. Create a 1-D tensor of 10 zeros (★☆☆)**
+
+```elixir
+Nx.broadcast(0, {10})
+```
+
+**3. How to find the number of elements in a tensor (★☆☆)**
+
+```elixir
+Nx.tensor([[1, 2, 3], [4, 5, 6]])
+|> Nx.size()
+```
+
+**4. How to find the memory size of any tensor (★☆☆)**
+
+```elixir
+Nx.tensor([[1, 2, 3], [4, 5, 6]])
+|> Nx.byte_size()
+```
+
+**5. How to get the documentation of the `add/2` function from the command line? (★☆☆)**
+
+```elixir
+# I don't know if this is possible.
+# iex> h Nx.add
+```
+
+**6. Create a tensor of zeros of size 10 but the fifth value which is 1 (★☆☆)**
+
+```elixir
+zeros = Nx.broadcast(0, {10})
+index = Nx.tensor([4])
+Nx.indexed_put(zeros, index, 1)
+```
+
+**7. Create a tensor with values ranging from 10 to 49 (★☆☆)**
+
+```elixir
+Nx.linspace(10, 49, n: 39, type: :s8)
+```
+
+**8. Reverse a tensor (first element becomes last) (★☆☆)**
+
+```elixir
+Nx.iota({10})
+|> Nx.reverse()
+```
+
+**9. Create a 3x3 tensor with values ranging from 0 to 8 (★☆☆)**
+
+```elixir
+Nx.iota({3, 3})
+```
+
+**10. Find indices of non-zero elements from [1,2,0,0,4,0] (★☆☆)**
+
+```elixir
+Nx.tensor([1, 2, 0, 0, 4, 0])
+|> Nx.not_equal(0)
+```
+
+## 11-20
+
+**11. Create a 3x3 identity tensor (★☆☆)**
+
+```elixir
+Nx.eye(3)
+```
+
+**12. Create a 3x3x3 tensor with random values (★☆☆)**
+
+```elixir
+key = Nx.Random.key(0)
+{random, _} = Nx.Random.normal(key, shape: {3, 3, 3})
+random
+```
+
+**13. Create a 10x10 tensor with random values and find the minimum and maximum values (★☆☆)**
+
+```elixir
+# This felt a little awkward. Is there a better way?
```

Comment:
Do you mean that using random felt awkward?

Unfortunately there's not a better way. The interface is as such because `Nx.Random` provides a stateless PRNG framework, which lends itself to being run alongside calculations on GPUs.

Reply:
Oh whoops, no this was a holdover from when I wasn't yet using `reduce_max`. I'll drop this comment.

Reply:
Oh and FWIW, I actually find the way `Nx.Random` works to make a lot of sense. It's got a very functional programming feel.

Reply:
[1475f49](https://github.com/elixir-nx/nx/pull/1363/commits/1475f4910701a8f2ed00dd5f07e67f9417909772)